### ISSUE TITLE
format k8s api server error message

### ISSUE THREAD

**Issue by cppforlife** [k14s/kapp#102]
currently kapp shows k8s api server error verbatim

> kapp: Error: Applying update job/pi (batch/v1) namespace: default: Updating resource job/pi (batch/v1) namespace: default: Job.batch "pi" is invalid: [spec.selector: Required value, spec.template.metadata.labels: Invalid value: map[string]string{"kapp.k14s.io/app":"1586905796363557000", "kapp.k14s.io/association":"v1.a4db8f96450049336d37eb62d798d883"}: `selector` does not match template `labels`, spec.selector: Invalid value: "null": field is immutable, spec.template: Invalid value: core.PodTemplateSpec{ObjectMeta:v1.ObjectMeta{Name:"", GenerateName:"", Namespace:"", SelfLink:"", UID:"", ResourceVersion:"", Generation:0, CreationTimestamp:v1.Time{Time:time.Time{wall:0x0, ext:0, loc:(*time.Location)(nil)}}, DeletionTimestamp:(*v1.Time)(nil), DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string{"kapp.k14s.io/app":"1586905796363557000", "kapp.k14s.io/association":"v1.a4db8f96450049336d37eb62d798d883"}, Annotations:map[string]string(nil), OwnerReferences:[]v1.OwnerReference(nil), Initializers:(*v1.Initializers)(nil), Finalizers:[]string(nil), ClusterName:"", ManagedFields:[]v1.ManagedFieldsEntry(nil)}, Spec:core.PodSpec{Volumes:[]core.Volume(nil), InitContainers:[]core.Container(nil), Containers:[]core.Container{core.Container{Name:"pi", Image:"perl", Command:[]string{"perl", "-Mbignum=bpi", "-wle", "print bpi(2000)"}, Args:[]string(nil), WorkingDir:"", Ports:[]core.ContainerPort(nil), EnvFrom:[]core.EnvFromSource(nil), Env:[]core.EnvVar(nil), Resources:core.ResourceRequirements{Limits:core.ResourceList(nil), Requests:core.ResourceList(nil)}, VolumeMounts:[]core.VolumeMount(nil), VolumeDevices:[]core.VolumeDevice(nil), LivenessProbe:(*core.Probe)(nil), ReadinessProbe:(*core.Probe)(nil), Lifecycle:(*core.Lifecycle)(nil), TerminationMessagePath:"/dev/termination-log", TerminationMessagePolicy:"File", ImagePullPolicy:"Always", SecurityContext:(*core.SecurityContext)(nil), Stdin:false, StdinOnce:false, TTY:false}}, RestartPolicy:"Never", TerminationGracePeriodSeconds:(*int64)(0xc00e08a8d0), ActiveDeadlineSeconds:(*int64)(nil), DNSPolicy:"ClusterFirst", NodeSelector:map[string]string(nil), ServiceAccountName:"", AutomountServiceAccountToken:(*bool)(nil), NodeName:"", SecurityContext:(*core.PodSecurityContext)(0xc012c22bd0), ImagePullSecrets:[]core.LocalObjectReference(nil), Hostname:"", Subdomain:"", Affinity:(*core.Affinity)(nil), SchedulerName:"default-scheduler", Tolerations:[]core.Toleration(nil), HostAliases:[]core.HostAlias(nil), PriorityClassName:"", Priority:(*int32)(nil), PreemptionPolicy:(*core.PreemptionPolicy)(nil), DNSConfig:(*core.PodDNSConfig)(nil), ReadinessGates:[]core.PodReadinessGate(nil), RuntimeClassName:(*string)(nil), EnableServiceLinks:(*bool)(nil)}}: field is immutable] (reason: Invalid)

would be nice if we added formatted version that may be omits some info:

```
kapp: Error: Applying update job/pi (batch/v1) namespace: default:
  Updating resource job/pi (batch/v1) namespace: default:
    Job.batch "pi" is invalid:
    - spec.selector: Required value
    - spec.template.metadata.labels: Invalid value: map[string]string{...}: `selector` does not match template `labels`
    - spec.selector: Invalid value: "null": field is immutable
    - spec.template: Invalid value: core.PodTemplateSpec{...}: field is immutable
```

**Comment by cppforlife** [comment]
i vaguely remember that some go objects were shrunk with "...", so it might be useful to actually find k8s code responsible for producing such error messages.

**Comment by cppforlife** [comment]
agg error:

https://github.com/kubernetes/kubernetes/blob/a5e6ac0a959b059513c1e7908fbb0713467839c4/staging/src/k8s.io/apimachinery/pkg/util/errors/errors.go#L64

individual validation errors:

https://github.com/kubernetes/kubernetes/blob/a5e6ac0a959b059513c1e7908fbb0713467839c4/staging/src/k8s.io/apimachinery/pkg/util/validation/field/errors.go#L175

**Comment by cppforlife** [comment]
pulled out logic into https://github.com/cppforlife/go-cli-ui/tree/master/errors

**Comment by cppforlife** [comment]
first cut of this was included in v0.26.0
